synergy-design-system/synergy-design-system
# feat: ✨ syn-select / syn-option / syn-optgroup

# Pull Request

## 📖 Description

This PR adds the following elements to our component-library:

- `<syn-select />`
- `<syn-option />`
- `<syn-optgroup />`

### 🎫 Issues

Closes #126

## 👩‍💻 Reviewer Notes

As discussed, we are adding multiple new components in this PR. Please have a look at storybook. For developers, also make sure the framework wrappers work as intended.

### Design:

- Please have a look at all paddings, as well as the icon sizes. Colors should match quite good, however the new keyboard state is NOT documented in Figma (as far as I know)

### Dev:

- Please also have a look at the provided tests.
- @mariohamann: Would need some feedback about the accessibility role for optgroup. It also errors out in the demos Accessibility panel (as it is missing the select, which was done intentional).

## 📑 Test Plan

- Check out the branch and prepare it (`pnpm i -r && pnpm build`)
- Start storybook (`cd packages/docs && pnpm start`)
  - Have a look at the new stories (syn-select, syn-option and syn-optgroup)
- Spin up the local framework demos and play around with them. They should appear in the debug output when changed.

## ✅ DoD
